### DISCUSSION TITLE
Error updating to the latest version

### DISCUSSION THREAD

**Discussion by thrangster** [homebridge-simplisafe3/homebridge-simplisafe3#338] [General]
Cannot update to the latest version 1.10.4 .

Ironically it mentions its a fix for some people who cannot update...

![Screen Shot 2022-04-22 at 1 39 11 PM](https://user-images.githubusercontent.com/68309317/164766119-bd10ae51-07a3-407f-86b3-0951a3fac2de.png)

**Comment by shamoon** [comment]
I think you just need to uninstall the old one and try again, the log says right there `ENOTEMPTY`

**Reply by thrangster** [reply]
Well its working fine.... if I uninstall the old one and install new, do I have to set everything up again?

**Reply by shamoon** [reply]
You should:

1. make sure you have a backup (ideally of Homebridge and your HomeKit install with something like Controller+).
2. Make sure **persistAccessories** is enabled, this will prevent removing the items (but again, have a backup in case).
3. Depending on the specifics of your installation you can usually uninstall something and re-install it and as long as you havent restarted homebridge youre ok because the plugin is already in memory. If you are using the UI make sure to check the leave config option, its probably best to do this via the command line.

Im not really sure why youre getting the error though, its not actually the same thing this release fixes.

**Reply by thrangster** [reply]
Maybe I’ll wait till the next update and see if it self corrects

**Comment by thrangster** [comment]
I can’t even uninstall…

[
<img width="832" alt="PNG image" src="https://user-images.githubusercontent.com/68309317/164873958-2c639d73-5f30-42df-af73-e1bd496f0195.png">

**Reply by shamoon** [reply]
Erm, well that certainly points to an issue on your system as opposed to the plug-in…

did you look around for solutions? https://stackoverflow.com/questions/43778883/npm-err-code-enotempty-while-npm-install

**Reply by thrangster** [reply]
All other modules work and update correctly

I’m not very technical - based on the error and the link you sent what should I try to uninstall ( what do I type in terminal?)

thanks

**Reply by shamoon** [reply]
Perhaps 
`sudo rm -rf /usr/local/lib/node modules/homebridge-simplisafe`

`sudo rm -rf /us/local/lib/node modules/.homebridge-simplisafe3-oalhgLLK`

I’m not sure if that second one actually exists  but either way it’s fine. That should remove them and then try to install again

are you using the Homebridge ui to install / update?

**Reply by thrangster** [reply]
Yes Homebridge UI to try an update or uninstall...

The above don't seem to to anything. I run them in the terminal, I immediately get a new command prompt and no feedback that anything was done

![image](https://user-images.githubusercontent.com/68309317/164894327-95149576-695e-4f57-828e-f35cb5dae537.png)

and still get the error

![image](https://user-images.githubusercontent.com/68309317/164894362-4861b2f1-e941-40da-b3d6-98dfd4b71a62.png)

While loaded it is no longer "working" - no accessories appear in Homebridge and No Response from devices already in Apple Home...

I cannot downgrade the version to see if they helps me then upgrade or uninstall....

This is crazy

All other plug in are working.... (TuyaSmart, Ring,

**Comment by thrangster** [comment]
I restored a backup from a few months back - reauthenticated, and at least got my accessories to appear again in HB and Home, but I still have the update and deleting issues above...

What are the commands to update or delete from Terminal if this might work?

Thanks

**Reply by shamoon** [reply]
Can you post the log file it mentioned eg `/home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log`

so after deleting those directories (`sudo rm…`) you then tried to install via command line or the UI? And you can install / upgrade / downgrade all other plugins except this one?

typically installation is performed via command line with `sudo npm install -g —unsafe-perm homebridge-simplisafe3` but I was thinking it might be better to let the UI do it for you (I’m unclear if that’s not working)

you could also ask on [Reddit](https://www.reddit.com/r/homebridge/) or [discord](https://discord.gg/kqNCe2D) btw, at the moment I don’t think this is related to the plug-in itself (you can’t even downgrade for example)

**Reply by thrangster** [reply]
Yes, all other plug ins are fine, and I can uninstall/install - simplisafe is also working but I cannot delete it or up/downgrade. It stopped working at one point add after the remove suggestions, but I restored from back up and rea-uthneticated and it started to work again - just cannot delete or upgrade. So pretty much limited to Simplisafe plug in

How do I get to the log?

I entered `/home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log` in terminal but get

`bash: /home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log: No such file or directory`

This is running in a Raspberry Pi... my computer is a mac...

Thanks

**Reply by shamoon** [reply]
So there was an error in what I said before because the text-recognition on my phone was incorrect. Incidentally, this is a note to say its always better to copy + paste the text from the terminal rather than send an image screenshot.

Try: `sudo rm -rf /usr/local/lib/node_modules/.homebridge-simplisafe3-oa0hgLLK` and then try again. You should double-check that command, like you should copy and paste the path from the terminal output. My phone incorrectly copied `/us/local` but just now I noticed on my Mac it used an O instead of 0 in the last part. Again, thats a reason to copy + paste when reporting stuff like this when reporting. But anyway, just to say when you run that command double check its the right path.

Also, the log files are files, so on the pi you'd need to find them or run `cat  /home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log`

**Reply by thrangster** [reply]
it says there is no log there?
```
pi@homebridge:/var/lib/homebridge $ cat /home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log
cat: /home/pi/.npm/_logs/2022-04-23T12_24_11_6907-debug-0.log: No such file or directory
```

**Reply by thrangster** [reply]
trying the rm suggestion....

**Reply by thrangster** [reply]
So it looks like the `rm -rf` command, followed by uninstalling the plug in from the plug in GUI, worked. It uninstalled, and I was then able to install the latest version in the same boot instance, so all the settings appear to be retained - 

But now all the Simplisafe devices show **NO RESPONSE** in Apple Home (they _do_ appear in Accessories in Homebridge and seem to be accurate as to the sensor state). I rebooted Homebridge, and also re-authenticated the plug in and rebooted, but no difference.

All other Homebridge plug-ins are appearing and functioning correctly in Apple Home...

![image](https://user-images.githubusercontent.com/68309317/164979800-e6cc0cff-f15d-4d92-9e7a-d1d9f0f15c59.png)

![image](https://user-images.githubusercontent.com/68309317/164979811-1cba08a7-751d-40f0-8c6b-e4cc4de34d4d.png)

Only Simplisafe in Home shows this...

![Screenshot 2022-04-24 at 9 59 28 AM](https://user-images.githubusercontent.com/68309317/164980179-29aaa422-31b7-4b29-bb97-0ce8d59209c3.jpeg)


Any thoughts here?

Thanks

**Reply by shamoon** [reply]
As I said this was never really an issue with the plug-in, the install got messed up for whatever reason. Now it’s working and you need to do some general troubleshooting, try restarting your hubs, etc but you may need to set things up again. That’s why I recommend having backups on the holebridge side and the HomeKit side (eg [Controller for HomeKit](https://controllerforhomekit.com/)). If you still cant figure it out please reach out on the Homebridge Discord or Reddit.
